TinyME-main/src/main/java/ir/ramtung/tinyme/domain/entity/Order.java

```diff
@@ -0,0 +1,107 @@
+package ir.ramtung.tinyme.domain.entity;
+
+import ir.ramtung.tinyme.messaging.request.EnterOrderRq;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.time.LocalDateTime;
+
+@Builder
+@EqualsAndHashCode
+@ToString
+@Getter
+public class Order {
+    protected long orderId;
+    protected Security security;
+    protected Side side;
+    protected int quantity;
+    protected int price;
+    protected Broker broker;
+    protected Shareholder shareholder;
+    @Builder.Default
+    protected LocalDateTime entryTime = LocalDateTime.now();
+    @Builder.Default
+    protected OrderStatus status = OrderStatus.NEW;
+
+    public Order(long orderId, Security security, Side side, int quantity, int price, Broker broker, Shareholder shareholder, LocalDateTime entryTime, OrderStatus status) {
+        this.orderId = orderId;
+        this.security = security;
+        this.side = side;
+        this.quantity = quantity;
+        this.price = price;
+        this.entryTime = entryTime;
+        this.broker = broker;
+        this.shareholder = shareholder;
+        this.status = status;
+    }
+
+    public Order(long orderId, Security security, Side side, int quantity, int price, Broker broker, Shareholder shareholder, LocalDateTime entryTime) {
+        this.orderId = orderId;
+        this.security = security;
+        this.side = side;
+        this.quantity = quantity;
+        this.price = price;
+        this.entryTime = entryTime;
+        this.broker = broker;
+        this.shareholder = shareholder;
+        this.status = OrderStatus.NEW;
+    }
+
+    public Order(long orderId, Security security, Side side, int quantity, int price, Broker broker, Shareholder shareholder) {
+        this(orderId, security, side, quantity, price, broker, shareholder, LocalDateTime.now());
+    }
+
+    public Order snapshot() {
+        return new Order(orderId, security, side, quantity, price, broker, shareholder, entryTime, OrderStatus.SNAPSHOT);
+    }
+
+    public Order snapshotWithQuantity(int newQuantity) {
+        return new Order(orderId, security, side, newQuantity, price, broker, shareholder, entryTime, OrderStatus.SNAPSHOT);
+    }
+
+    public boolean matches(Order other) {
+        if (side == Side.BUY)
+            return price >= other.price;
+        else
+            return price <= other.price;
+    }
```

Comment:
```suggestion
    public boolean matches(Order other) {
        if(side.opposite() == other.side()){
            if (side == Side.BUY)
                return price >= other.price;
            else
                return price <= other.price;
        }
        else
            throw new IllegalArgumentException("The sides are similar");
    }
```

Reply:
It might be better to split the "matches" method into separate methods named "matchesBuy" and "matchesSell".